py/objstr.c

```diff
@@ -123,7 +123,7 @@ STATIC void str_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t
     bool is_bytes = true;
     #endif
     if (kind == PRINT_RAW || (!MICROPY_PY_BUILTINS_STR_UNICODE && kind == PRINT_STR && !is_bytes)) {
-        mp_printf(print, "%.*s", str_len, str_data);
+        mp_print_strn(print, (char *)str_data, str_len, 0, 0, 0);
```

Comment:
What's the reason for this change?  If this code is going to be changed it might be better to go straight to: `print->print_strn(print->data, str_data, str_len)`.

Reply:
The strings that are passed into str_print can contain embedded null-bytes, eg. print('%c' % False).
With the behavior of mp_vprintf corrected, these strings would now be truncated because strlen() is performed on them, stopping at the embedded null-byte.
Because we already know the string-length we should be using a function that actually outputs all bytes.

I agree that we should go straight to print->print_strn(...).
All of the other logic in mp_print_strn handles with the width and padding, which is not required here.

Two unit-tests captured this behavior, which alerted me to change this function.
- basics/string_format_module.py
- basics/string_strip.py

Reply:
Ok, thanks, sounds good.

If you can change it to use `print->print_strn` then this PR will be good to go.

Reply:
Also, please rebase on latest master, then hopefully the CI failure will be fixed.